manim/mobject/geometry.py

```diff
@@ -838,6 +838,21 @@ def get_unit_vector(self):
     def get_angle(self):
         return angle_of_vector(self.get_vector())
 
+    def get_projection(self, point):
```

Comment:
This is a generally useful function and shouldn't be tied to `Line` imo. Can we move this into `manim\utils\space_ops.py`? It would look like [this](https://github.com/abhi3019/manim-projects/blob/2e831d76ecaa06b37beda00ea0d4c767be24f372/custom_functions/manim.py#L5-L12)